db/src/main/java/net/bible/android/database/BookmarksDao.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2020 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
+ *
+ * This file is part of And Bible (http://github.com/AndBible/and-bible).
+ *
+ * And Bible is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with And Bible.
+ * If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+package net.bible.android.database
+
+import androidx.room.Dao
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Update
+import org.crosswire.jsword.versification.BibleBook
+
+@Dao
+interface BookmarkDao {
+    @Query("SELECT * from Bookmark")
+    fun allBookmarks(): List<BookmarkEntities.Bookmark>
+
+    @Query("SELECT * from Label")
+    fun allLabels(): List<BookmarkEntities.Label>
+
+    @Query("SELECT * from bookmark")
+    fun allBookmarksSorted(): List<BookmarkEntities.Bookmark>
+
+    @Query("SELECT * from Bookmark where id = :bookmarkId")
+    fun bookmarkById(bookmarkId: Long): BookmarkEntities.Bookmark
+
+    @Query("SELECT * from Bookmark where id IN (:bookmarkIds)")
+    fun bookmarksByIds(bookmarkIds: LongArray): List<BookmarkEntities.Bookmark>
```

Comment:
You can also do `List<Long>` I believe. I've done `List<Int>`

Reply:
Yes; I just used this as it is simpler because the data is coming in this form where this function is used